trunk/Bricksmith/Source/Widgets/ScopeBar.m

```diff
@@ -49,27 +49,32 @@ - (id)initWithFrame:(NSRect)frame
 //				here. Furthermore, the appearance will be degenerate on Tiger 
 //				because I do not feel like pounding out the manual CoreGraphics 
 //				calls which NSGradient is using. Sooner or later we will be 
-//				dumping Tiger and then none of it will matter! 
+//				dumping Tiger and then none of it will matter!
+//
+//				[1-9-2022 @undecoded-coder]: Commenting this out for now as it
+//				works poorly with macOS Dark Mode. Furthermore, I don't think
+//				this styling is relevant to modern macOS (Big Sur and later)
+//				anymore.
 //
 //==============================================================================
-- (void) drawRect:(NSRect)rect
-{
-	NSColor         *topColor       = [NSColor colorWithCalibratedWhite:0.91 alpha:1.0];
-	NSColor         *bottomColor    = [NSColor colorWithCalibratedWhite:0.82 alpha:1.0];
-	NSColor         *borderColor    = [NSColor colorWithCalibratedWhite:0.58 alpha:1.0];
-	NSRect          bounds          = [self bounds];
-	NSBezierPath    *bottomLine     = [NSBezierPath bezierPath];
-	NSGradient      *gradient       = [[[NSGradient alloc] initWithStartingColor:bottomColor endingColor:topColor] autorelease];
-	
-	[gradient drawInRect:bounds angle:90];
-	
-	// bottom line border
-	[bottomLine moveToPoint:NSMakePoint( NSMinX(bounds) - 0.5, NSMinY(bounds) )];
-	[bottomLine lineToPoint:NSMakePoint( NSMaxX(bounds) - 0.5, NSMinY(bounds) )];
-	
-	[borderColor set];
-	[bottomLine stroke];
-	
-}//end drawRect:
+//- (void) drawRect:(NSRect)rect
+//{
+//	NSColor         *topColor       = [NSColor colorWithCalibratedWhite:0.91 alpha:1.0];
+//	NSColor         *bottomColor    = [NSColor colorWithCalibratedWhite:0.82 alpha:1.0];
+//	NSColor         *borderColor    = [NSColor colorWithCalibratedWhite:0.58 alpha:1.0];
+//	NSRect          bounds          = [self bounds];
+//	NSBezierPath    *bottomLine     = [NSBezierPath bezierPath];
+//	NSGradient      *gradient       = [[[NSGradient alloc] initWithStartingColor:bottomColor endingColor:topColor] autorelease];
+//
+//	[gradient drawInRect:bounds angle:90];
+//
+//	// bottom line border
+//	[bottomLine moveToPoint:NSMakePoint( NSMinX(bounds) - 0.5, NSMinY(bounds) )];
+//	[bottomLine lineToPoint:NSMakePoint( NSMaxX(bounds) - 0.5, NSMinY(bounds) )];
+//
+//	[borderColor set];
+//	[bottomLine stroke];
+//
+//}//end drawRect:
```

Comment:
Why does Apple do stupid stuff like this? It looks like not only have they _still_ not added any API for drawing these bars, they aren't even consistent in their own applications anymore. Boo.

Ordinarily, I would say delete the code instead of leaving it commented out. But since that's all this class actually did, I guess it's okay if you leave it just so it isn't so confusing.

Reply:
Removed references in storyboard (nib) files to the ScopeBar class since it isn't doing anything useful anymore. Wasn't sure if it would be fine to delete the source files or not?

Reply:
It needs to be removed from `LDrawDocument.xib` as well.

Then just remove the entire class. (Verify full removal by doing a Find All for "ScopeBar".)

Reply:
The references in the `xib` files (`PartBrowser.xib` as well) were removed already, and there are no other references.